lms/static/scripts/frontend_apps/components/dashboard/DashboardOrganizationFilters.tsx

```diff
@@ -0,0 +1,111 @@
+import { MultiSelect } from '@hypothesis/frontend-shared';
+
+import type { Assignment, Course, Student } from '../../api-types';
+import { useConfig } from '../../config';
+import { useAPIFetch } from '../../utils/api';
+
+export type DashboardFiltersProps = {
+  selectedCourses: Course[];
+  onCoursesChange: (newCourses: Course[]) => void;
+  selectedAssignments: Assignment[];
+  onAssignmentsChange: (newAssignments: Assignment[]) => void;
+  selectedStudents: Student[];
+  onStudentsChange: (newStudents: Student[]) => void;
+};
+
+export default function DashboardOrganizationFilters({
+  selectedCourses,
+  onCoursesChange,
+  selectedAssignments,
+  onAssignmentsChange,
+  selectedStudents,
+  onStudentsChange,
+}: DashboardFiltersProps) {
+  const { dashboard } = useConfig(['dashboard']);
+  const { routes } = dashboard;
+
+  const courses = useAPIFetch<{ courses: Course[] }>(routes.courses);
+  const assignments = useAPIFetch<{ assignments: Assignment[] }>(
+    routes.assignments,
+  );
+  const students = useAPIFetch<{ students: Student[] }>(routes.students);
+
+  return (
+    <div className="flex gap-2 md:w-1/2">
+      <MultiSelect
+        disabled={courses.isLoading}
+        value={selectedCourses}
+        onChange={onCoursesChange}
+        buttonContent={
+          courses.isLoading ? (
+            <>Loading...</>
```

Comment:
I think we should change this to something more subtle like leaving it blank or use "..." if loading. That will make the transition less noticeable.